discovery/service.go

```diff
@@ -137,20 +141,22 @@ type Discovery struct {
 // Remote channel announcements should contain the announcement proof and be
 // fully validated.
 func (d *Discovery) ProcessRemoteAnnouncement(msg lnwire.Message,
-	src *btcec.PublicKey) error {
+	src *btcec.PublicKey) chan error {
```

Comment:
The tests should instead wrap the method call in a goroutine to perform the select at a higher level. In most instances golang pacakge should expose a _synchronous_  API, and leave it to the callers to add additional async semantics on top.